src/message.rs

```diff
@@ -199,9 +258,33 @@ impl From<Value7> for u8 {
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct Value14(u8, u8);
 
+impl Value14 {
+    /// Create a new `Value14`
+    ///
+    /// # Arguments
+    /// * `val` - the value
+    ///
+    /// # Note
+    /// * The `val` will be clamped so it is in the 0..127 valid range
+    ///
+    pub const fn new(val: i16) -> Self {
+        let value = if val <= -8192i16 {
+            -8192i16
+        } else if val >= 8191i16 {
+            8191i16
+        } else {
+            val
+        }
+        .saturating_add(8192i16) as u16;
```

Comment:
over here you could use `val.clamp`, and you could get rid of the saturating add too. I think overflowing add is cheaper (please correct me if I'm wrong here) and because you already clamped the value this will never overflow anyway. You could add a comment to indicate that.

Reply:
`.clamp` isn't `const` but you're right, `overflow_add` should actually be safe then